hierarchy/store.go

```diff
@@ -53,59 +58,62 @@ func (store *Store) BuildHierarchy(instanceID, codeListID, dimensionName string)
 		"dimension_name": dimensionName,
 	}
 
+	db := &database{5, nil}
+
 	log.Debug("requesting neo4j connection from the connection pool", logData)
-	connection, err := store.pool.OpenPool()
+	var err error
+	db.conn, err = store.pool.OpenPool()
 	if err != nil {
 		return err
 	}
 
-	defer connection.Close()
+	defer db.conn.Close()
 
-	err = createInstanceHierarchyConstraints(1, 5, connection, instanceID, dimensionName)
+	err = db.createInstanceHierarchyConstraints(1, instanceID, dimensionName)
```

Comment:
Rather than always having to pass in 1, it would be a lot cleaner (in my opinion) to have a second method signature which defaults off to 1:
```
func (db *database) createInstanceHierarchyConstraints(instanceID, dimensionName string) error {
    return db.createInstanceHierarchyConstraints(1, instanceID)
}
```

Reply:
I think to make this change we'd have to create a lot of function signatures, and function names, that don't mean anything so i feel like it might make the code harder to follow if we made this change

also we have another instance of a similar pattern with hardcoded first attempts here:https://github.com/ONSdigital/dp-observation-importer/blob/cmd-develop/observation/store.go#L101

i agree that it feels a bit weird, but to me seem sthe most sane approach, particularly in an unexported method :)